pkg/v1/tkg/client/client_helper.go

```diff
@@ -0,0 +1,113 @@
+		packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: kappPkgShortName, Namespace: namespace},
+			Spec: kapppkgv1alpha1.PackageSpec{RefName: kappPkgName, Version: kappPkgVersion}})
+	}
+	if clusterBootstrap.Spec.CNI != nil {
+		kappPkgShortName, kappPkgName, kappPkgVersion := getPackageDetailsFromCBS(clusterBootstrap.Spec.CNI.RefName)
+		packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: kappPkgShortName, Namespace: namespace},
+			Spec: kapppkgv1alpha1.PackageSpec{RefName: kappPkgName, Version: kappPkgVersion}})
+	}
+	if clusterBootstrap.Spec.CSI != nil {
+		kappPkgShortName, kappPkgName, kappPkgVersion := getPackageDetailsFromCBS(clusterBootstrap.Spec.CSI.RefName)
+		packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: kappPkgShortName, Namespace: namespace},
+			Spec: kapppkgv1alpha1.PackageSpec{RefName: kappPkgName, Version: kappPkgVersion}})
+	}
+	if clusterBootstrap.Spec.CPI != nil {
+		kappPkgShortName, kappPkgName, kappPkgVersion := getPackageDetailsFromCBS(clusterBootstrap.Spec.CPI.RefName)
+		packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: kappPkgShortName, Namespace: namespace},
+			Spec: kapppkgv1alpha1.PackageSpec{RefName: kappPkgName, Version: kappPkgVersion}})
+	}
+
+	return packages
+}
+
+// getPackageDetailsFromCBS parse given clusterbootstrap package reference and returns package short name, package name and version details
+func getPackageDetailsFromCBS(CBSRefName string) (pkgShortName, pkgName, pkgVersion string) {
+	pkgShortName = strings.Split(CBSRefName, ".")[0]
+	pkgName = strings.Join(strings.Split(CBSRefName, ".")[0:4], ".")
+	pkgVersion = strings.Join(strings.Split(CBSRefName, ".")[4:], ".")
+	return
+}
+
+// MonitorAddonsCorePackageInstallation monitors addon's core packages (kapp, cni, csi and cpi) and returns error if any while monitoring packages or any packages are not installed successfully. First it monitors kapp package in management cluster then it monitors other core packages in workload cluster.
+func MonitorAddonsCorePackageInstallation(regionalClusterClient clusterclient.Client, workloadClusterClient clusterclient.Client, packages []kapppkgv1alpha1.Package, packageInstallTimeout time.Duration) error {
+	if len(packages) == 0 {
+		return nil
+	}
+	var corePackages []string
+	var kappPackage []string
+	for _, p := range packages {
+		if strings.Contains(p.ObjectMeta.Name, "kapp-controller") {
+			kappPackage = append(kappPackage, p.ObjectMeta.Name)
+		} else {
+			corePackages = append(corePackages, p.ObjectMeta.Name)
+		}
+	}
+	if len(kappPackage) > 0 {
+		err := WaitForPackagesInstallation(regionalClusterClient, kappPackage, packages[0].ObjectMeta.Namespace, packageInstallTimeout)
+		if err != nil {
+			return err
+		}
+	}
+	return WaitForPackagesInstallation(workloadClusterClient, corePackages, packages[0].ObjectMeta.Namespace, packageInstallTimeout)
```

Comment:
What is the need to split packages into kappPackages and corePackages and waiting for them separately? Can't we wait for them in a single `WaitForPackagesInstallation` method?